<launch>
   <!-- Frontier Parametes -->
  <arg name="rate" default="1"/>
  <arg name="namespace" default="drone"/>
  <arg name="odometry" default="/mavros/local_position/odom"/>
  <arg name="cloud_in" default="/camera/depth/points"/>
  <arg name="config_filename" default="$(find uav_frontier_exploration_3d)/config/UAV.yaml"/>



  <!-- Clustering Parameters -->
  <arg name="obstacle_voxel" default="obstacle_voxel"/>
  <arg name="free_voxel" default="free_voxel" />
  <arg name="uav_frontier" default="frontier_cells_vis_array"/>
  


  <!-- Launch UAV frontier exporation -->
  <group ns="$(arg namespace)">
    <node name="frontier_server_node" pkg="uav_frontier_exploration_3d" type="frontier_server_node" output="screen">
      <param name="rate" value="$(arg rate)"/>
      <param name="exploration_config_filename" value="$(arg config_filename)"/>
      <remap from="cloud_in" to="$(arg cloud_in)"/>
      <remap from="odometry" to="$(arg odometry)"/>
      <remap from="namespace" to="$(arg namespace)"/>
    </node>

     <!-- Launch UAV clustering -->
    <node pkg="uav_frontier_exploration_3d" type="ClusteringVoxel.py" name="ClusteringVoxel" output="screen">
      <param name="quantile" value="0.05" /> 
      <param name="radius_threshold" value="0.5" /> 
      <param name="frame" value="odom" />
      <param name="marker_ns" value="uav_cluster" />
      <remap from="obstacle_voxel" to="$(arg obstacle_voxel)"/>
      <remap from="free_voxel" to="$(arg free_voxel)" />
      <remap from="uav_frontier" to="$(arg uav_frontier)"/>
    </node>
   </group>



</launch>

